apps/web/scripts/kernel/handlers/kernel.ts

```diff
@@ -4,34 +4,39 @@ import { RequestType } from '@walless/messaging';
 import { handle } from '../utils/coordinator';
 import { getNetwork } from '../utils/handler';
 
-import {
-	handleConnect,
-	handleInstallLayout,
-	handleRequestPayload,
-} from './common';
+import * as commonHandler from './common';
 import * as solanaHandler from './solanaHandler';
 import * as suiHandler from './suiHandler';
-import * as tezosHanlder from './tezosHandler';
+import * as tezosHandler from './tezosHandler';
 
 export const onKernelMessage: MessengerCallback = async (payload, channel) => {
 	const { type, requestId } = payload;
 
 	if (requestId) {
 		let handleMethod;
 		let requirePrivateKey = true;
+		let requireUserAction = true;
 
 		if (type === RequestType.REQUEST_CONNECT) {
-			handleMethod = handleConnect;
+			handleMethod = commonHandler.handleConnect;
 			requirePrivateKey = false;
 		} else if (type == RequestType.GET_ENDPOINT_ON_SOLANA) {
 			solanaHandler.getEndpoint(payload, channel);
 			requirePrivateKey = false;
 		} else if (type === RequestType.REQUEST_PAYLOAD) {
-			handleMethod = handleRequestPayload;
+			handleMethod = commonHandler.handleRequestPayload;
 			requirePrivateKey = false;
 		} else if (type === RequestType.INSTALL_LAYOUT) {
-			handleMethod = handleInstallLayout;
+			handleMethod = commonHandler.handleInstallLayout;
 			requirePrivateKey = false;
+		} else if (type === RequestType.CHECK_INSTALLED_LAYOUT) {
+			handleMethod = commonHandler.handleCheckInstalledLayout;
+			requirePrivateKey = false;
+			requireUserAction = false;
```

Comment:
like previous comment, this request may get all installed layouts, checking existing of a specific layout might be made by user (or from calling side)

Reply:
yes, this method could be refactored in the future if use case increased